fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java

```diff
@@ -0,0 +1,229 @@
+                throw new ExternalAssetOwnerInitiateTransferException(
+                        "External asset owner transfer is already in BUYBACK state for this loan.");
+            }
+        }
+    }
+
+    private ExternalAssetOwnerTransfer parseJson(Long loanId, String apiRequestBodyAsJson, ExternalId externalLoanId) {
+        ExternalAssetOwnerTransfer externalAssetOwnerTransfer = new ExternalAssetOwnerTransfer();
+
+        validateRequestBody(apiRequestBodyAsJson);
+        final JsonElement json = fromApiJsonHelper.parse(apiRequestBodyAsJson);
+
+        ExternalAssetOwner owner = getOwner(json);
+        externalAssetOwnerTransfer.setOwnerId(owner.getId());
+        externalAssetOwnerTransfer.setOwner(owner);
+        externalAssetOwnerTransfer.setExternalId(getTransferExternalIdFromJson(json));
+        externalAssetOwnerTransfer.setStatus(ExternalTransferStatus.PENDING.name());
+        externalAssetOwnerTransfer.setPurchasePriceRatio(getPurchasePriceRatioFromJson(json));
+        externalAssetOwnerTransfer.setSettlementDate(getSettlementDateFromJson(json));
+        externalAssetOwnerTransfer.setEffectiveDateFrom(ThreadLocalContextUtil.getBusinessDate());
+        externalAssetOwnerTransfer.setEffectiveDateTo(LocalDate.of(9999, 12, 31));
+        externalAssetOwnerTransfer.setLoanId(loanId);
+        externalAssetOwnerTransfer.setExternalLoanId(externalLoanId);
+        return externalAssetOwnerTransfer;
+    }
+
+    private void validateRequestBody(String apiRequestBodyAsJson) {
+        final Set<String> requestParameters = new HashSet<>(
+                Arrays.asList(ExternalTransferRequestParameters.SETTLEMENT_DATE, ExternalTransferRequestParameters.OWNER_EXTERNAL_ID,
+                        ExternalTransferRequestParameters.TRANSFER_EXTERNAL_ID, ExternalTransferRequestParameters.PURCHASE_PRICE_RATIO,
+                        ExternalTransferRequestParameters.DATEFORMAT, ExternalTransferRequestParameters.LOCALE));
+        final Type typeOfMap = new TypeToken<Map<String, Object>>() {}.getType();
+        fromApiJsonHelper.checkForUnsupportedParameters(typeOfMap, apiRequestBodyAsJson, requestParameters);
+    }
+
+    private LocalDate getSettlementDateFromJson(JsonElement json) {
+        String dateFormat = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.DATEFORMAT, json);
+        String locale = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.LOCALE, json);
+        return fromApiJsonHelper.extractLocalDateNamed(ExternalTransferRequestParameters.SETTLEMENT_DATE, json, dateFormat,
+                JsonParserHelper.localeFromString(locale));
+    }
+
+    private ExternalId getTransferExternalIdFromJson(JsonElement json) {
+        String transferExternalId = fromApiJsonHelper.extractStringNamed(ExternalTransferRequestParameters.TRANSFER_EXTERNAL_ID, json);
+        return new ExternalId(transferExternalId);
```

Comment:
Please use the producer + if externalId is not provided we need to generate one.